packages/ui/src/components/shadcn/ui/table.tsx

```diff
@@ -163,7 +159,16 @@ const TableCaption = React.forwardRef<
   HTMLTableCaptionElement,
   React.HTMLAttributes<HTMLTableCaptionElement>
 >(({ className, ...props }, ref) => (
-  <caption ref={ref} className={cn('mt-4 text-sm text-foreground-muted', className)} {...props} />
+  <caption
+    ref={ref}
+    className={cn(
+      'border-t', // TableCaption is aligned by parent Table at caption-bottom
+      'p-4 text-sm text-foreground-muted', // Match styling of TableCell
+      className
+    )}
+    // Should only contain inline elements
+    {...props}
```

Comment:
As explained in comments:

- It always needs a `border-t`
- It should match the styling of `TableCell`